src/templates/TemplateData.ts

```diff
@@ -202,16 +231,56 @@ function getRuntimeKey(baseKey: string, runtime: ProjectRuntime): string {
     return runtime === ProjectRuntime.one ? baseKey : `${baseKey}.${runtime}`;
 }
 
-async function requestFunctionPortal<T>(hostname: string, subpath: string, runtime: string, param?: string): Promise<T> {
-    const options: request.OptionsWithUri = {
+async function cachedTemplatesAreCurrent(globalState?: vscode.Memento): Promise<boolean> {
+    if (globalState) {
+        for (const key of Object.keys(ProjectRuntime)) {
+            const runtime: ProjectRuntime = <ProjectRuntime>ProjectRuntime[key];
+            const funcJsonOptions: request.OptionsWithUri = {
+                method: 'GET',
+                uri: funcCliFeedUrl,
+                headers: {
+                    'User-Agent': 'Mozilla/5.0' // Required otherwise we get Unauthorized
+                }
+            };
+
+            const cliFeedJson: cliFeedJsonResponse = <cliFeedJsonResponse>JSON.parse(await <Thenable<string>>request(funcJsonOptions).promise());
+            const currentRelease: string = <string>cliFeedJson.tags[runtime].release;
+            const cachedRelease: string | undefined = globalState.get(`${runtime}-release`);
+            if (!cachedRelease || parseInt(cachedRelease, 10) < parseInt(currentRelease, 10)) {
+                // templates are not up-to-date and need to be downloaded/extracted
+                return false;
+            }
+
+            // make sure templates were cached
+            const cachedResources: object | undefined = globalState.get<object>(getRuntimeKey(resourcesKey, runtime));
+            const cachedTemplates: object[] | undefined = globalState.get<object[]>(getRuntimeKey(templatesKey, runtime));
+            const cachedConfig: object | undefined = globalState.get<object>(getRuntimeKey(configKey, runtime));
+            if (!cachedResources || !cachedTemplates || !cachedConfig) {
+                return false;
+            }
+        }
+        // only return true if ALL runtimes are current
+        return true;
+    }
+    return false;
+}
+
+async function getCurrentReleaseTemplates(runtime: string): Promise<string> {
+    const funcJsonOptions: request.OptionsWithUri = {
         method: 'GET',
-        uri: `https://${hostname}/api/${subpath}?runtime=${runtime}&${param}`,
+        uri: funcCliFeedUrl,
         headers: {
             'User-Agent': 'Mozilla/5.0' // Required otherwise we get Unauthorized
         }
     };
 
-    return <T>(JSON.parse(await <Thenable<string>>request(options).promise()));
+    const cliFeedJson: cliFeedJsonResponse = <cliFeedJsonResponse>JSON.parse(await <Thenable<string>>request(funcJsonOptions).promise());
+    const currentRelease: string = <string>cliFeedJson.tags[runtime].release;
+    const urlWithPythonTemplates: string = 'https://functionscdn.azureedge.net/public/TemplatesApi/2.0.0-beta-10180.zip';
+    // temp ternary for dev purposes
+    const templateUrl: string = runtime === ProjectRuntime.two ? urlWithPythonTemplates : <string>cliFeedJson[currentRelease].templateApiZip;
```

Comment:
Not sure what "temp ternary for dev purposes" means. I suggest getting rid of this or filing an issue on github to address before the release